pkgs/games/rbdoom-3-bfg/default.nix

```diff
@@ -0,0 +1,65 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, directx-shader-compiler
+, libGLU
+, openal
+, SDL2
+, vulkan-headers
+, vulkan-loader
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "rbdoom-3-bfg";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "RobertBeckebans";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-r/dvTirgFXdBJ+Gjl6zpHoGCTPoo0tRmOCV9oCdnltI=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    directx-shader-compiler
+  ];
+
+  buildInputs = [
+    libGLU
+    openal
+    SDL2
+    vulkan-headers
+    vulkan-loader
+    zlib
+  ];
+
+  cmakeDir = "../neo";
+  cmakeFlags = [
+    "-DFFMPEG=OFF"
+    "-DBINKDEC=ON"
+    "-DUSE_SYSTEM_ZLIB=ON"
```

Comment:
Do you think you can enable `-DUSE_SYSTEM_LIBPNG=ON`, `-DUSE_SYSTEM_LIBJPEG=ON`, `-DUSE_SYSTEM_LIBGLEW=ON`, and `-DUSE_SYSTEM_RAPIDJSON=ON` too?

Thank you for this!

Reply:
Wasn't able to get it to build with `-DUSE_SYSTEM_LIBJPEG=ON` due to `jpegint.h` not being found. I believe the fix is in RBDOOM-3-BFG master branch, but for now it doesn't work.

Reply:
Do you think you could add something like this to pull the commit from master branch that fixes it?

This diff builds for me

```diff
diff --git a/pkgs/games/rbdoom-3-bfg/default.nix b/pkgs/games/rbdoom-3-bfg/default.nix
index d2c91a0395f..375fec174e4 100644
--- a/pkgs/games/rbdoom-3-bfg/default.nix
+++ b/pkgs/games/rbdoom-3-bfg/default.nix
@@ -1,10 +1,12 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, fetchpatch
 , cmake
 , directx-shader-compiler
 , libGLU
 , libpng
+, libjpeg
 , openal
 , rapidjson
 , SDL2
@@ -25,6 +27,14 @@ stdenv.mkDerivation rec {
     fetchSubmodules = true;
   };
 
+  patches = [
+    (fetchpatch {
+      name = "remove-jpeg_internals-define.patch";
+      url = "https://github.com/RobertBeckebans/RBDOOM-3-BFG/commit/de6ab9d31ffcd6eba26df69f8c77da38a0ab4722.diff";
+      hash = "sha256-3XbWmQtY/8a90IqDtN5TNT5EOa+i5mFOH+H9tuZqTmU=";
+    })
+  ];
+
   nativeBuildInputs = [
     cmake
     directx-shader-compiler
@@ -33,6 +43,7 @@ stdenv.mkDerivation rec {
   buildInputs = [
     libGLU
     libpng
+    libjpeg
     openal
     rapidjson
     SDL2
@@ -47,6 +58,7 @@ stdenv.mkDerivation rec {
     "-DBINKDEC=ON"
     "-DUSE_SYSTEM_LIBGLEW=ON"
     "-DUSE_SYSTEM_LIBPNG=ON"
+    "-DUSE_SYSTEM_LIBJPEG=ON"
     "-DUSE_SYSTEM_RAPIDJSON=ON"
     "-DUSE_SYSTEM_ZLIB=ON"
   ];
```